hdKeysGenerator/export/exporter.go

```diff
@@ -0,0 +1,128 @@
+package export
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-tools-go/hdKeysGenerator/common"
+)
+
+// ArgsNewExporter holds arguments for creating an exporter
+type ArgsNewExporter struct {
+	ActualShard    common.OptionalUint32
+	ProjectedShard common.OptionalUint32
+	StartIndex     int
+	NumKeys        int
+	Format         string
+	OutputFile     string
+}
+
+type exporter struct {
+	actualShard    common.OptionalUint32
+	projectedShard common.OptionalUint32
+	startIndex     int
+	numKeys        int
+	format         string
+	outputFile     string
+}
+
+// NewExporter creates a new exporter
+func NewExporter(args ArgsNewExporter) (*exporter, error) {
+	return &exporter{
+		actualShard:    args.ActualShard,
+		projectedShard: args.ProjectedShard,
+		startIndex:     args.StartIndex,
+		numKeys:        args.NumKeys,
+		format:         args.Format,
+		outputFile:     args.OutputFile,
+	}, nil
+}
+
+// ExportKeys exports the generated keys to a file
+func (e *exporter) ExportKeys(keys []common.GeneratedKey) error {
+	log.Info("exporting:",
+		"numKeys", len(keys),
+		"formatType", e.format,
+	)
+
+	err := e.saveKeysFile(keys)
+	if err != nil {
+		return err
+	}
+
+	err = e.saveMetadataFile()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *exporter) saveKeysFile(keys []common.GeneratedKey) error {
+	formatter, err := e.getFormatter()
+	if err != nil {
+		return err
+	}
+
+	text, err := formatter.toText(keys)
+	if err != nil {
+		return err
+	}
+
+	err = e.saveFile(e.outputFile, text)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (e *exporter) getFormatter() (formatter, error) {
```

Comment:
please call this directly on the constructor as to early check if the provided formatter string is valid or not. The `exporter` struct can hold the formatter instance rather than the formatter string
A separate factory method would have been great